src/document/advanced.rs

```diff
@@ -9,37 +13,39 @@ use crate::{
 };
 use itertools::EitherOrBoth;
 
+/// IronOxide Advanced Document Operations
+///
+/// # Key Terms
+/// - ID     - The ID representing a document. It must be unique within the document's segment and will **not** be encrypted.
```

Comment:
Do we need to mention ID and Name here? They aren't used in this trait, right?

Reply:
Well `id` and `name` are fields of `DocumentEncryptOpts`, which is shared for both kinds of encryption even if it maybe shouldn't be. 
And `id` is returned as part of `DocumentEncryptUnmanagedResult` and `DocumentDecryptUnmanagedResult`.
So it seems like we should at least have some mention here.

Reply:
Since they aren't directly applicable, I would remove them. Maybe some of the explanation about uniqueness and encryption should be on `DocumentEncryptOps`?

Reply:
There is a sentence about ID uniqueness there. I've added the part about not encrypting them there and removed those two bullet points from `DocumentAdvancedOps`.